File: Benchmarks/enterprise/python/flask/testcode/benchmark_test_79945.py
# SPDX-License-Identifier: Apache-2.0
from flask import request, jsonify


def coalesce_blank(value):
    return value or ''

def BenchmarkTest79945():
    referer_value = request.headers.get('Referer', '')
    data = coalesce_blank(referer_value)
    return jsonify({'error': 'An internal error occurred'}), 500

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest79945.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest79945 {

    @PostMapping(path="/BenchmarkTest79945", consumes="text/plain")
    public void BenchmarkTest79945(@RequestBody String rawBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String rawData = rawBody != null ? rawBody : "";
        String data = "[%s]".formatted(rawData);
        java.util.regex.Pattern inputPattern = java.util.regex.Pattern.compile("[a-zA-Z0-9_-]+");
        if (inputPattern.matcher(data).find()) {
            response.setHeader("X-Validated-Input", data);
        }
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}
